lib/rubygems/specification_policy.rb

```diff
@@ -367,8 +369,14 @@ def validate_licenses_length
   def validate_licenses
     licenses = @specification.licenses
 
+    num_classes = licenses.map(&:class).uniq.size
+
+    if !licenses.empty? && num_classes != 1
+      error "licenses must be of the same class, detected #{num_classes} classes"
+    end
+
```

Comment:
Given my other comment, can we simply ignore nil if it is present, but not print the unwanted warning if it is explicitly set to nil? There is no conflict having nil in an array if we don't try to overload nil with meaning.